src/migrations/20231122121456-dedupe-any-duplicate-permissions.js

```diff
@@ -0,0 +1,72 @@
+exports.up = function (db, cb) {
+    db.runSql(
+        `
+-- STEP 1: Update references in the role_permission table, setting the permission to the lowest ID for each duplicate permission
+
+WITH Duplicates AS (
+    SELECT
+        MIN(id) AS min_id,
+        permission
+    FROM
+        permissions
+    GROUP BY
+        permission
+    HAVING
+        COUNT(*) > 1
+)
+
+UPDATE
+    role_permission
+SET
+    permission_id = d.min_id
+FROM
+    Duplicates d
+JOIN
+    permissions p ON d.permission = p.permission
+WHERE
+    role_permission.permission_id = p.id
+AND
+    role_permission.permission_id != d.min_id;
+
+
+-- STEP 2: Delete redundant role_permission entries
+
+DELETE FROM role_permission
+WHERE ctid IN (
+    SELECT ctid
+    FROM (
+        SELECT ctid, ROW_NUMBER() OVER (PARTITION BY role_id, permission_id, environment ORDER BY created_at) as rn
+        FROM role_permission
+    ) t
+    WHERE t.rn > 1
+);
+
+-- STEP 3: Delete the duplicate permissions, keeping the ones with the lowest ID
+
+WITH Duplicates AS (
+    SELECT
+        MIN(id) AS min_id,
+        permission
+    FROM
+        permissions
+    GROUP BY
+        permission
+    HAVING
+        COUNT(*) > 1
+)
+
+DELETE FROM permissions
+WHERE id NOT IN (
+    SELECT min_id FROM Duplicates
+)
+AND permission IN (
+    SELECT permission FROM Duplicates
+);
```

Comment:
Can we also apply a unique constraint to the name?

Reply:
This is currently part of https://github.com/Unleash/unleash/pull/5398/files#diff-c979c34ca172be9fd07e04898e1256e9e305dd0384f74d888c862809413ed69aR4-R6
Do you think it belongs here instead, since this is the exact time we clean this up?

Reply:
No, I'm fine as long as it's considered! Unblocking